src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -271,6 +266,11 @@ public Image<TPixel> Decode<TPixel>(BufferedReadStream stream, CancellationToken
                                 previousFrameControl = currentFrameControl;
                             }
 
+                            if (frameCount == this.maxFrames)
```

Comment:
maybe use `>= this.maxFrames` here to be extra safe?

Reply:
Yeah, multiple `fcTL` in a row might cause this. Will update.

Reply:
Done